src/client/Microsoft.Identity.Client/Cache/CacheSessionManager.cs

```diff
@@ -121,35 +121,26 @@ private async Task RefreshCacheForReadOperationsAsync(CacheEvent.TokenTypes cach
                             using (_requestParams.RequestContext.CreateTelemetryHelper(cacheEvent))
                             {
                                 string key = CacheKeyFactory.GetKeyFromRequest(_requestParams);
+                                var args = new TokenCacheNotificationArgs(
+                                      TokenCacheInternal,
+                                      _requestParams.AppConfig.ClientId,
+                                      _requestParams.Account,
+                                      hasStateChanged: false,
+                                      isApplicationCache: TokenCacheInternal.IsApplicationCache,
+                                      suggestedCacheKey: key,
+                                      hasTokens: TokenCacheInternal.HasTokensNoLocks(),
```

Comment:
Does this change the caching operation behavior since the `HasTokens` is called once now? Would `OnBeforeAccessAsync` ever change this value?

Reply:
Im not sure it would make any difference here. The `HasTokensNoLocks` method ultimately checks to see if there is any tokens by pulling from the cache and unless there is a change to the cache (`hasStateChanged`) the value should remain the same for the duration of this method. But @bgavrilMS can confirm.

Reply:
Better to be on the safe side, I'll revert this small change.